lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useRef, useContext, useEffect, useState } from 'react';
+import classnames from 'classnames';
+
+import { nxToastPropTypes, NxToastProps } from './types';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+import ToastContext from './contexts';
+import { toastTypeMap } from './toastTypeMapping';
+
+const NxToast = (props: NxToastProps) => {
+  const { toastId, className, message, type, toastContainerRef, ...otherProps } = props,
+      toastClass = toastTypeMap[type].class,
+      toastIcon = toastTypeMap[type].icon,
+      toastIconLabel = toastTypeMap[type].iconLabel,
+      [animate, setAnimate] = useState(false),
+      classes = classnames('nx-toast', 'animate', className, toastClass,
+          {'slide-in': animate},
+          {'slide-out': !animate}
```

Comment:
I don't think I understand why it's necessary to have two of these classes.  I would assume there's only two animations states: it's either slid up and transparent or it's slid into place and opaque.  Toggling between those two can be done by adding/removing a single class

Reply:
> I updated the code so it only uses on the slide-in class. The slide-out class was removed: [8ffc769](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/8ffc7693bf6a8fb6b79b243f92e6c674babd8dc5)

Looks good, but I also question the name of that class.  Classes are states (in this case) and `slide-in` sounds more like a state _transition_ than an actual state.  How about naming it `visible` or `nx-toast--visible`?

Reply:
Renamed here: [9ee7b42](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/9ee7b425191200363802d36a521ca3823a418e69)